From d0c5c5d29d4cc299649573fed3a2e112e828412d Mon Sep 17 00:00:00 2001
From: Dylan Hutchison <dhutchis@cs.washington.edu>
Date: Thu, 8 Jun 2017 16:22:28 -0700
Subject: [PATCH] [SYSTEMML-1668] Fix rbind size estimates in BinaryOp

In BinaryOp.inferOutputCharacteristics, RBIND is not checked and CBIND is checked twice. Fix the second case to refer to RBIND.

Closes #533.
---
 src/main/java/org/apache/sysml/hops/BinaryOp.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main/java/org/apache/sysml/hops/BinaryOp.java b/src/main/java/org/apache/sysml/hops/BinaryOp.java
index 17a099f78f8..ccf73c4157f 100644
--- a/src/main/java/org/apache/sysml/hops/BinaryOp.java
+++ b/src/main/java/org/apache/sysml/hops/BinaryOp.java
@@ -844,7 +844,7 @@ protected long[] inferOutputCharacteristics( MemoTable memo )
 			if( ldim1 > 0 || ldim2 > 0 || lnnz >= 0 )
 				return new long[]{ldim1, ldim2, lnnz};
 		}
-		else if( op== OpOp2.CBIND ) {
+		else if( op == OpOp2.RBIND ) {
 			long ldim1 = -1, ldim2 = -1, lnnz = -1;
 			
 			if( mc[0].colsKnown() || mc[1].colsKnown() )
